pkg/git/source.go

```diff
@@ -0,0 +1,15 @@
+package git
```

Comment:
Any particular reason not to use git source from the api package?

Reply:
The reason was that I didn't realize that the GitSource is already part of this repo :-). Thanks for pointing it out.
The actual integration with the controller (and reading of the secrets) will be part of different PR